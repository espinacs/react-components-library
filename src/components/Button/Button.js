// eslint-disable jsx-a11y/click-events-have-key-events jsx-a11y/no-static-element-interactions
import React from 'react';
import { PropTypes } from 'prop-types';

require('./styles.css');

const Button = (props) => {
  const classType = props.type ? ` button-type-${props.type}` : '';
  return (
    <button data-testid="button" className={`button${classType}`} onClick={props.onPress}>
      {props.children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node,
  type: PropTypes.string,
  onPress: PropTypes.func.isRequired,
};

Button.defaultProps = {
  children: null,
  type: null,
};

export default Button;
