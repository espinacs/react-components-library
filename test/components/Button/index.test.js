import React from 'react';
import { renderIntoDocument, cleanup } from 'react-testing-library';
import Button from '../../../src/components/Button';

const mockClick = jest.fn();
afterEach(cleanup);

test('mounts empty, renders null', () => {
  const { container } = renderIntoDocument(<Button onPress={mockClick} />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders something inside', () => {
  const expected = 'test';
  const { container } = renderIntoDocument(<Button onPress={mockClick}>{expected}</Button>);
  expect(container.firstChild).toMatchSnapshot();
});


test('can be clicked', () => {
  const expected = 'test';
  const { getByText } = renderIntoDocument(<Button onPress={mockClick}>{expected}</Button>);
  getByText(expected).click();
  expect(mockClick).toHaveBeenCalledTimes(1);
});

test('does apply light style', () => {
  const expected = 'test';
  const { container } = renderIntoDocument(<Button onPress={mockClick} type="light">{expected}</Button>);
  expect(container.firstChild).toMatchSnapshot();
});
