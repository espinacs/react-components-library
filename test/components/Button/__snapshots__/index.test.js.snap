// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does apply light style 1`] = `
<button
  class="button button-type-light"
  data-testid="button"
>
  test
</button>
`;

exports[`mounts empty, renders null 1`] = `
<button
  class="button"
  data-testid="button"
/>
`;

exports[`renders something inside 1`] = `
<button
  class="button"
  data-testid="button"
>
  test
</button>
`;
